src/rules/use-layers.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * @fileoverview Rule to require layers in CSS.
+ * @author Nicholas C. Zakas
+ */
+
+//-----------------------------------------------------------------------------
+// Rule Definition
+//-----------------------------------------------------------------------------
+
+export default {
+	meta: {
+		type: /** @type {const} */ ("problem"),
+
+		docs: {
+			description: "Require use of layers",
+			recommended: true,
+		},
+
+		schema: {
+			type: "array",
+			items: {
+				type: "object",
+				properties: {
+					allowUnnamedLayers: {
+						type: "boolean",
+					},
+					requireImportLayers: {
+						type: "boolean",
+					},
+					layerNamePattern: {
+						type: "string",
+					},
+				},
+				additionalProperties: false,
+			},
+			minItems: 0,
+			maxItems: 1,
+		},
```

Comment:
I think the `schema` property should be an array to match the `JSONSchema4` types. That's how most of our rules are defined.

```suggestion
		schema: [
			{
				type: "array",
				items: {
					type: "object",
					properties: {
						allowUnnamedLayers: {
							type: "boolean",
						},
						requireImportLayers: {
							type: "boolean",
						},
						layerNamePattern: {
							type: "string",
						},
					},
					additionalProperties: false,
				},
				minItems: 0,
				maxItems: 1,
			},
		],
```

The TypeScript error message indicates the problem (somehow) here:

          The types of 'meta.schema' are incompatible between these types.
            Type '{ type: string; items: { type: string; properties: { allowUnnamedLayers: { type: string; }; requireImportLayers: { type: string; }; layerNamePattern: { type: string; }; }; additionalProperties: boolean; }; minItems: number; maxItems: number; }' is not assignable to type 'false | JSONSchema4 | JSONSchema4[] | undefined'.

It's unobvious, but `schema` needs to be an array to satisfy `JSONSchema4` (unless it's defined with `anyOf`, `oneOf`, etc.).

Reply:
Ah gotcha, thanks!